src/jbms/argparse.cpp

```diff
@@ -19,15 +19,108 @@ namespace argparse {
 
 using std::weak_ptr;
 
-string repr(string_view s) {
-  // FIXME: make this escape bad characters
-  string x;
-  x += '\'';
-  x += string(s);
-  x += '\'';
-  return x;
+void _chr_cstrliteral(unsigned char ch, char *buffer, size_t buflen) {
+  /**
+   * @brief Escapes a single character @p ch and appends its escaped value onto
+   * the buffer.
+   * @param ch Character to escape (if a control character).
+   * @param buffer Buffer to store the escaped value in.
+   * @param buflen Buffer length.
+   */
+  if (buflen < 2) {
+    *buffer = '\0';  // only null-terminator can be stored
+  } else if (isprint(ch) && ch != '\'' && ch != '\"' && ch != '\\' &&
+             ch != '\?') {
+    sprintf(
+        buffer, "%c",
+        ch);  // opposite of control characters can be saved just as they are
+  } else if (buflen < 3) {
+    *buffer = '\0';  // just add a null-terminator; don't need to go further
+  } else {
+    switch (ch) {
+      case '\a':
+        strcpy(buffer, "\\a");
+        break;
+      case '\b':
+        strcpy(buffer, "\\b");
+        break;
+      case '\f':
+        strcpy(buffer, "\\f");
+        break;
+      case '\n':
+        strcpy(buffer, "\\n");
+        break;
+      case '\r':
+        strcpy(buffer, "\\r");
+        break;
+      case '\t':
+        strcpy(buffer, "\\t");
+        break;
+      case '\v':
+        strcpy(buffer, "\\v");
+        break;
+      case '\\':
+        strcpy(buffer, "\\\\");
+        break;
+      case '\'':
+        strcpy(buffer, "\\'");
+        break;
+      case '\"':
+        strcpy(buffer, "\\\"");
+        break;
+      case '\?':
+        strcpy(buffer, "\\\?");
+        break;
+      default:
+        if (buflen < 5)
+          *buffer = '\0';
+        else
+          sprintf(buffer, "\\%03o",
+                  ch);  // this handles control characters that are not
+                        // printable and does not meet any case
+        break;
+    }
+  }
+}
+
+void _str_cstrliteral(const char *str, char *buffer, size_t buflen) {
+  /**
+   * @brief Escapes the special string @p str and stores inside of the @p
+   * buffer.
+   * @param str Special characters string.
+   * @param buffer Buffer to store the escaped value in.
+   * @param buflen Length of the buffer.
+   */
+  size_t blen;
+  unsigned char ch;
+  while ((ch = (unsigned char)*str++) != '\0') {
+    _chr_cstrliteral(ch, buffer, buflen);
```

Comment:
This program works for almost every control character but is broken for embedded `NULL` byte. @jbms What do you think? Are you going to merge it? Because the only thing that is in compromise is just a `NULL` byte.